### ISSUE TITLE
Confusion around the term "zone"

### ISSUE THREAD

**Issue by maelvls** [Venafi/vcert#257]
I have been around Venafi TPP for a while, and I just realized today that "Zone" is a term that only ever exist in the `vcert` code. It does not exist in the TPP documentation, nor in the VaaS documentation. Also, I was not able to find what "Zone" meant for VaaS.

From what I can gather:

**TPP**

In TPP, a zone is the DN (distinguished name) of a given policy folder with the leading `\VED\Policy\` optionally removed. For example, the following screenshot shows the policy "folder" that has the DN `\VED\Policy\Jetstack`:

![Screenshot from 2022-09-12 11-28-51](https://user-images.githubusercontent.com/2195781/189620297-28526ff5-ebee-4b43-8643-befd806ae1b1.png)

There are three equivalent ways of writing the same "zone" (as defined in [`getPolicyDN`](https://github.com/Venafi/vcert/blob/b89f255872d0253f67902cd9d5f270386a8d9ad6/pkg/venafi/tpp/tpp.go#L557-L568)):

```
\VED\Policy\Jetstack
\Jetstack
Jetstack
```

Internally, they are all converted into the same DN:

```
\VED\Policy\Jetstack
```

**VaaS**

In VaaS, the zone is the concatenation of the application name and the template name. For example, in the screenshot, the application is cert-manager and the template is e2e, so the zone is `cert-manager\e2e`:

  <img alt="" src="https://user-images.githubusercontent.com/2195781/189379681-77c16132-6fa3-43f1-addd-36cf92ba56b8.png" width="60%">

I am confused as to why the term "Zone" is used for both TPP and VaaS when their semantics are so different in each? Maybe a solution could be to warn people in the readme about "Zone", the fact that it is a term only used in the context of `vcert`, and that it means a different thing depending on the service you are using.

**Comment by tr1ck3r** [comment]
@maelvls VCert's primary objective was, and for the most part still remains, to abstract away the differences between the TPP and VaaS API layers and provide a consistent experience regardless of the Venafi product that provides the certificate services.  TPP and VaaS, of course, not only have different API layers but they also took different approaches to processing certificates.

We adopted the "zone" term to generalize the configuration that represents the certificate policy, the issuing certificate authority, and any product-specific associations that accompany a certificate request.  So, for TPP a zone is a policy folder (distinguished name of a policy object, optionally omitting the `\VED\Policy` prefix) and for VaaS a zone is the names of an application and that application's alias for an issuing template (the policy and CA that governs the issuance of a certificate request).  

Basically "zone" is the unfortunate consequence of lack of standards between two products that provide the same service.  What it refers to is clear to any Venafi user but it does require documentation to explain.
